navigator/runtime-experimental/src/main/java/com/freeletics/mad/navigator/compose/NavHost.kt

```diff
@@ -59,6 +68,31 @@ public fun NavHost(
     }
 }
 
+@Composable
+private fun SystemBackHandling(executor: MultiStackNavigationExecutor) {
+    val backPressedDispatcher = LocalOnBackPressedDispatcherOwner.current
+    val callback = remember(executor) {
+        object : OnBackPressedCallback(executor.canNavigateBack.value) {
+            override fun handleOnBackPressed() {
+                executor.navigateBack()
+            }
+
+        }
+    }
+    LaunchedEffect(executor, callback) {
+        snapshotFlow { executor.canNavigateBack.value }
+            .distinctUntilChanged()
+            .collect { callback.isEnabled = it }
+    }
+    DisposableEffect(backPressedDispatcher, callback) {
+        backPressedDispatcher!!.onBackPressedDispatcher.addCallback(callback)
```

Comment:
I think this should fail early in case `LocalOnBackPressedDispatcherOwner` can't be obtained

Reply:
[5fccfae](https://github.com/freeletics/mad/pull/381/commits/5fccfaee40b6c895c20c0fecf03ea14220c56695)